modules/stochastic_tools/src/userobject/PolynomialChaos.C

```diff
@@ -0,0 +1,240 @@
+
+    // Evaluate polynomials to avoid duplication
+    for (unsigned int d = 0; d < _ndim; ++d)
+      for (unsigned int i = 0; i < _order; ++i)
+        poly_val(d, i) = _poly[d]->compute(i, data[d]);
+
+    // Loop over coefficients
+    for (unsigned int i = 0; i < _ncoeff; ++i)
+    {
+      Real val = _values[p - _sampler->getLocalRowBegin()];
+      // Loop over parameters
+      for (std::size_t d = 0; d < _ndim; ++d)
+        val *= poly_val(d, _tuple[i][d]);
+
+      if (_quad_sampler)
+        val *= _quad_sampler->getQuadratureWeight(p);
+      _coeff[i] += val;
+    }
+  }
+}
+
+void
+PolynomialChaos::finalize()
+{
+  gatherSum(_coeff);
+
+  if (!_quad_sampler)
+    for (unsigned int i = 0; i < _ncoeff; ++i)
+      _coeff[i] /= _sampler->getNumberOfRows();
+}
+
+void
+PolynomialChaos::threadJoin(const UserObject & y)
+{
+  const PolynomialChaos & pps = static_cast<const PolynomialChaos &>(y);
+  for (unsigned int i = 0; i < _ncoeff; ++i)
+    _coeff[i] += pps._coeff[i];
+}
+
+Real
+PolynomialChaos::evaluate(const std::vector<Real> & x) const
+{
+
+  mooseAssert(x.size() == _ndim, "Number of inputted parameters does not match PC model.");
+
+  DenseMatrix<Real> poly_val(_ndim, _order);
+
+  // Evaluate polynomials to avoid duplication
+  for (unsigned int d = 0; d < _ndim; ++d)
+    for (unsigned int i = 0; i < _order; ++i)
+      poly_val(d, i) = _poly[d]->compute(i, x[d], /*normalize =*/false);
+
+  Real val = 0;
+  for (unsigned int i = 0; i < _ncoeff; ++i)
+  {
+    Real tmp = _coeff[i];
+    for (unsigned int d = 0; d < _ndim; ++d)
+      tmp *= poly_val(d, _tuple[i][d]);
+    val += tmp;
+  }
+
+  return val;
+}
+
+Real
+PolynomialChaos::computeStandardDeviation() const
+{
+  Real var = 0;
+  for (unsigned int i = 1; i < _ncoeff; ++i)
+  {
+    Real norm = 1.0;
+    for (std::size_t d = 0; d < _ndim; ++d)
+      norm *= _poly[d]->innerProduct(_tuple[i][d]);
+    var += _coeff[i] * _coeff[i] * norm;
+  }
+
+  return std::sqrt(var);
+}
+
+Real
+PolynomialChaos::powerExpectation(const unsigned int n, const bool distributed) const
+{
+  std::vector<StochasticTools::WeightedCartesianProduct<unsigned int, Real>> order;
+  order.reserve(_ndim);
+  std::vector<std::vector<Real>> c_1d(n, std::vector<Real>(_coeff.begin() + 1, _coeff.end()));
+  for (unsigned int d = 0; d < _ndim; ++d)
+  {
+    std::vector<std::vector<unsigned int>> order_1d(n, std::vector<unsigned int>(_ncoeff - 1));
+    for (unsigned int i = 1; i < _ncoeff; ++i)
+      for (unsigned int m = 0; m < n; ++m)
+        order_1d[m][i - 1] = _tuple[i][d];
+    order.push_back(StochasticTools::WeightedCartesianProduct<unsigned int, Real>(order_1d, c_1d));
+  }
+
+  unsigned int n_local = order[0].numRows();
+  unsigned int st_local = 0;
+  unsigned int end_local = n_local;
+  if (distributed)
+    MooseUtils::linearPartitionItems(
+        order[0].numRows(), n_processors(), processor_id(), n_local, st_local, end_local);
```

Comment:
@zachmprince - This linearPartitionItems function has three output parameters (the last three). These are all declared as dof_id_type, which varies for 32 versus 64-bit builds. You need to declare these types as `dof_id_type` to avoid compile issues with the 64-bit build. You don't need to initialize them as they are output only parameters, at the very lest don't bother trying to set them to something intelligent.

Reply:
@permcody I switched to using `dof_id_type`. I'm not sure what you mean by "set them to something intelligent" though. I would like to set them just in case `distributed == false`. What is the procedure for updating this PR? Should I create a new PR? Also, is there any way to test on a 64-bit build just in case I missed something else?

Reply:
Nevermind, Andrew came by and answered my questions.

Reply:
Yeah, I see the branch case now. Setting them to appropriate values is correct.